src/main/java/loci/common/Location.java

```diff
@@ -778,11 +829,31 @@ public boolean isAbsolute() {
    * @return true if this pathname exists and represents a directory.
    * @see java.io.File#isDirectory()
    */
-  public boolean isDirectory() {
+  public boolean isDirectory() throws IOException {
```

Comment:
See https://web-proxy.openmicroscopy.org/east-ci/job/BIOFORMATS-merge/jdk=JDK8,label=testintegration/298/console 

```
[ERROR] /home/omero/workspace/BIOFORMATS-merge/jdk/JDK8/label/testintegration/bio-formats-build/bioformats/components/formats-bsd/src/loci/formats/FilePattern.java:[842,27] unreported exception java.io.IOException; must be caught or declared to be thrown
```

This contract might need to be propagated in a couple of places

Reply:
See https://github.com/openmicroscopy/bioformats/pull/3247